src/process_utility.c

```diff
@@ -2659,7 +2696,20 @@ timescaledb_ddl_command_start(
 		return;
 	}
 
-	if (!process_ddl_command_start(&args))
+	/*
+	 * Process Utility/DDL operation locally then pass it for execution
+	 * on any remote servers.
+	 */
+	args.hcache = ts_hypertable_cache_pin();
+
+	handled = process_ddl_command_start(&args);
+
+	if (ts_cm_functions->ddl_command_start)
+		ts_cm_functions->ddl_command_start(&args);
+
+	ts_cache_release(args.hcache);
```

Comment:
This should not be released here since `args` is passed to `prev_ProcessUtility` below. Also agree with @cevian that we probably want to do _all_ processing here first before calling the hook.